content/sensu-go/5.0/getting-started/sample-app.md

```diff
@@ -0,0 +1,363 @@
+---
+title: "Get started with the Sensu sample app"
+linkTitle: "Sample app"
+description: "Container and application monitoring with Sensu"
+weight: 3
+version: "5.0"
+product: "Sensu Go"
+menu:
+  sensu-go-5.0:
+    parent: getting-started
+---
+
+In this tutorial, we'll deploy a dummy app with Kubernetes and monitor it with Sensu.
+
+The dummy app has three endpoints: `/` returns the local hostname, `/metrics` returns app Prometheus metric data, `/healthz` returns the boolean app health state, and `POST /healthz` toggles the app health state.
+
+- [Prerequisites](#prerequisites)
+- [Setup](#setup)
+- [Multitenancy](#multitenancy)
+- [Deploying Sensu agents and InfluxDB](#deploying-sensu-agents-and-influxdb)
+- [Monitoring an app](#monitoring-an-app)
+	- [Create a Sensu pipeline to Slack](#create-a-sensu-pipeline-to-slack)
+	- [Create a Sensu service check to monitor the status of the dummy app](#create-a-sensu-service-check-to-monitor-the-status-of-the-dummy-app)
+- [Collecting app metrics](#collecting-app-metrics)
+	- [Create a Sensu pipeline to InfluxDB](#create-a-sensu-pipeline-to-influxdb)
+	- [Create a Sensu service check to collect Prometheus metrics](#create-a-sensu-service-check-to-collect-prometheus-metrics)
+	- [Use Grafana to visualize metrics](#use-grafana-to-visualize-metrics)
+- [Collecting Kubernetes metrics](#collecting-kubernetes-metrics)
+- [Next steps](#next-steps)
+
+## Prerequisites
+
+The sample app requires Kubernetes and a Kubernetes Ingress controller.
+Most hosted Kubernetes offerings, such as GKE, include a Kubernetes Ingress controller.
+
+## Setup
+
+**1. Clone the sample app.**
+
+{{< highlight shell >}}
+git clone git@github.com:sensu/sensu-kube-demo.git
+
+cd sensu-kube-demo
+{{< /highlight >}}
+
+**2. Create Kubernetes Ingress Resources.**
+
+{{< highlight shell >}}
+kubectl create -f go/ingress-nginx/ingress/sensu-go.yaml
+{{< /highlight >}}
+
+**3. Deploy kube-state-metrics.**
+
+{{< highlight shell >}}
+git clone git@github.com:kubernetes/kube-state-metrics.git
+
+kubectl apply -f kube-state-metrics/kubernetes
+{{< /highlight >}}
+
+**4. Open your `/etc/hosts` file and add the following hostnames.**
+
+{{< highlight shell >}}
+127.0.0.1       sensu.local webui.sensu.local sensu-enterprise.local dashboard.sensu-enterprise.local
+127.0.0.1       influxdb.local grafana.local dummy.local
+{{< /highlight >}}
+
+**5. Install sensuctl.**
+
+Jump over to the [sensuctl installation guide](https://docs.sensu.io/sensu-go/5.0/installation/install-sensu/#install-sensuctl), and follow the instructions to install sensuctl on Windows, macOS, or Linux.
+
+**6. Deploy two instances of a dummy app behind a load balancer.**
+
+{{< highlight shell >}}
+kubectl apply -f go/deploy/dummy.yaml
+{{< /highlight >}}
+
+We can test the dummy app using the API.
+
+{{< highlight shell >}}
+curl -i http://dummy.local
```

Comment:
Bumping up against this failing. But I'm using minikube here, so there's probably some stuff I've got to work out. All the instructions up until this point work.

Reply:
As I understand it, the dummy service is being assigned a default servicetype of "ClusterIP". This just means that no external traffic is allowed--only inter-pod/service traffic. It's possible that the nginx ingress rules are supposed to account for this, but I'm not familiar enough to really fully understand what those are doing. It's worth noting that I _think_ that the service file needs to be applied here as well, as that creates a loadbalancer service which exposes the respective services to traffic.

I was able to get a curlable dummy service by changing servicetype to nodeport, which does expose traffic, but I think this might be a bit of a workaround,given my previous statement.